weave/flow/model.py

```diff
@@ -32,20 +37,18 @@ def predict(self, input_data: str) -> dict:
     # TODO: should be infer: Callable
 
     def get_infer_method(self) -> Callable:
-        for infer_method_names in ("predict", "infer", "forward"):
-            infer_method = getattr(self, infer_method_names, None)
-            if infer_method:
+        for name in INFER_METHOD_NAMES:
+            if infer_method := getattr(self, name, None):
                 return infer_method
-        raise ValueError(
-            f"Model {self} does not have a predict, infer, or forward method."
+        raise MissingInferenceMethodError(
+            f"Missing a method with name like ({INFER_METHOD_NAMES})"
         )
 
 
 def get_infer_method(model: Model) -> Callable:
-    for infer_method_names in ("predict", "infer", "forward"):
-        infer_method = getattr(model, infer_method_names, None)
-        if infer_method:
+    for name in INFER_METHOD_NAMES:
+        if (infer_method := getattr(model, name, None)) is not None:
             return infer_method
-    raise ValueError(
-        f"Model {model} does not have a predict, infer, or forward method."
+    raise MissingInferenceMethodError(
+        f"Missing a method with name like ({INFER_METHOD_NAMES})"
```

Comment:
```suggestion
        f"Missing a method with name in ({INFER_METHOD_NAMES})"
```

I think "like" is a bit too broad, we need it to explicitly be included right?